### PR TITLE
sql/mysql: minor changes for AUTO_INCREMENT migration

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: sql/mysql/inspect.go [hunk 1]
```diff
@@ -494,6 +514,27 @@ func (i *inspect) myDefaultExpr(c *schema.Column, x, extra string) schema.Expr {
 	return &schema.Literal{V: quote(x)}
 }
 
+// parseColumn returns column parts, size and signed-info from a MySQL type.
+func parseColumn(typ string) (parts []string, size int64, unsigned bool, err error) {
```
Comment (a8m) [inline sql/mysql/inspect.go:518]: old code; moved to the bottom in the same file.

File: sql/mysql/diff.go [hunk 2]
```diff
@@ -161,6 +164,22 @@ func (d *diff) charsetChange(from, top, to []schema.Attr) schema.Change {
 	return noChange
 }
 
+// autoIncChange returns the schema change for changing the AUTO_INCREMENT
+// attribute in case it is not the default.
+func (d *diff) autoIncChange(from, to []schema.Attr) schema.Change {
+	var fromA, toA AutoIncrement
+	// The table is empty and AUTO_INCREMENT was not configured. This can happen
+	// because older versions of MySQL (< 8.0) stored the AUTO_INCREMENT counter
```
Comment (rotemtam) [inline sql/mysql/diff.go:172]: wow, didnt know this could happen

**Review by yonidavidson — APPROVED** [review]

File: sql/mysql/inspect.go [hunk 3]
```diff
@@ -426,40 +427,27 @@ func (i *inspect) tableNames(ctx context.Context, schema string, opts *schema.In
...
 		// The column has an expression default value
 		// and it is handled in Driver.addColumn.
 	case autoIncrement:
 		a := &AutoIncrement{}
-		// Reference to the table attribute if exists, as
-		// there can be only one auto_increment in a table.
-		sqlx.Has(t.Attrs, a)
+		// A table can have only one AUTO_INCREMENT column. If it was returned as NULL
+		// from INFORMATION_SCHEMA, it is due to information_schema_stats_expiry and we
+		// need to extract it from the 'CREATE TABLE' command.
```
Comment (rotemtam) [inline sql/mysql/inspect.go:442]: 🤯

**Review by rotemtam — APPROVED** [review]
Well done!

**Pull request merged**